programs/mpl-core/src/processor/add_external_plugin.rs

```diff
@@ -11,10 +23,71 @@ pub(crate) struct AddExternalPluginV1Args {
 }
 
 pub(crate) fn add_external_plugin<'a>(
-    _accounts: &'a [AccountInfo<'a>],
-    _args: AddExternalPluginV1Args,
+    accounts: &'a [AccountInfo<'a>],
+    args: AddExternalPluginV1Args,
 ) -> ProgramResult {
-    Err(MplCoreError::NotAvailable.into())
+    let ctx = AddExternalPluginV1Accounts::context(accounts)?;
+
+    // Guards.
+    assert_signer(ctx.accounts.payer)?;
+    let authority = resolve_authority(ctx.accounts.payer, ctx.accounts.authority)?;
+
+    if ctx.accounts.system_program.key != &solana_program::system_program::ID {
+        return Err(MplCoreError::InvalidSystemProgram.into());
+    }
+
+    if let Some(log_wrapper) = ctx.accounts.log_wrapper {
+        if log_wrapper.key != &spl_noop::ID {
+            return Err(MplCoreError::InvalidLogWrapperProgram.into());
+        }
+    }
+
+    if let Key::HashedAssetV1 = load_key(ctx.accounts.asset, 0)? {
+        msg!("Error: Adding plugin to compressed is not available");
+        return Err(MplCoreError::NotAvailable.into());
+    }
+
+    // let (asset, plugin_header, plugin_registry) = fetch_core_data::<AssetV1>(ctx.accounts.asset)?;
+
+    let validation_ctx = PluginValidationContext {
+        self_authority: &Authority::UpdateAuthority,
+        authority_info: authority,
+        resolved_authorities: None,
+        new_owner: None,
+        target_plugin: None,
+    };
```

Comment:
Ok so could you confirm, the reason we don't have to add target_external_plugin to the validation struct yet is because there's nothing yet like PermanentFreezeDelegate where the target plugin is used to self-validate?

Like right now only update authority can add an external plugin without any plugin-imposed restrictions?

Reply:
It's more that I haven't gotten to validation yet. I'm trying to stick with the add plugin scope so the context hasn't needed to be expanded yet.